romeo/asset-contract/contracts/asset.clar

```diff
@@ -67,17 +75,19 @@
     (tree-depth uint)
     (block-header (buff 80)))
     (begin
-        (try! (is-contract-owner))
+        (asserts! (is-contract-owner) err-forbidden)
+        (asserts! (> amount u0) err-bad-request)
         (try! (verify-txid-exists-on-burn-chain withdraw-txid burn-chain-height merkle-proof tx-index tree-depth block-header))
+        (try! (ft-burn? sbtc amount owner))
         (print {notification: "burn", payload: withdraw-txid})
-        (ft-burn? sbtc amount owner)
+    		(ok true)
     )
 )
 
 ;; #[allow(unchecked_data)]
 (define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
 	(begin
-		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-not-token-owner)
+		(asserts! (> amount u0) err-bad-request)
```

Comment:
The (ft-transfer? ...) function used below has also has a handful of built-in errors found [here](https://docs.stacks.co/docs/clarity/language-functions#ft-transfer). 

One of these built-in/global errors is specifically for sending a non-positive value. It's more clear for protocol developers if we throw that message instead of a custom check here. This line can/should be removed.

Reply:
gotcha